plugins/file_chooser/gtk/file_chooser_plugin.cc

```diff
@@ -0,0 +1,198 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "include/file_chooser/file_chooser_plugin.h"
+
+#include <flutter_linux/flutter_linux.h>
+#include <gtk/gtk.h>
+
+// See channel_controller.dart for documentation.
+const char kChannelName[] = "flutter/filechooser";
+const char kShowOpenPanelMethod[] = "FileChooser.Show.Open";
+const char kShowSavePanelMethod[] = "FileChooser.Show.Save";
+const char kInitialDirectoryKey[] = "initialDirectory";
+const char kInitialFileNameKey[] = "initialFileName";
+const char kAllowedFileTypesKey[] = "allowedFileTypes";
+const char kConfirmButtonTextKey[] = "confirmButtonText";
+const char kAllowsMultipleSelectionKey[] = "allowsMultipleSelection";
+const char kCanChooseDirectoriesKey[] = "canChooseDirectories";
+
+struct _FlFileChooserPlugin {
+  GObject parent_instance;
+
+  // Connection to Flutter engine.
+  FlMethodChannel* channel;
+
+  // Window to attach dialog to.
+  GtkWindow* window;
```

Comment:
Same comment here as in the other PR about keeping the registrar rather than the window.